src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,67 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> implementations = initImplementations();
+
+    private Map<Class<?>, Class<?>> initImplementations() {
+        return Map.of(
+                FileReaderService.class, FileReaderServiceImpl.class,
+                ProductService.class, ProductServiceImpl.class,
+                ProductParser.class, ProductParserImpl.class
+        );
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
+    private boolean isComponentAnnotated(Class<?> clazz) {
+        return clazz.isAnnotationPresent(Component.class);
+    }
+
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!isComponentAnnotated(interfaceClazz)) {
+            throw new RuntimeException("Missing Component annotation");
+        }
```

Comment:
interface class should not be annotated with component, you should check this annotation presence on implementation class